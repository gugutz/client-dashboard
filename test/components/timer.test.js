import React from 'react'
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import Timer from '../../src/components/timer'

describe('<Timer /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Timer seconds={12} onResetClick={jest.fn()}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should have a click button', () => {
    const handleResetClick = jest.fn()
    const wrapper = mount(
      <Timer seconds={12} onResetClick={handleResetClick}/>
    )
    wrapper.find('button').simulate('click')
    expect(handleResetClick.mock.calls.length).toBe(1)
  })
})
