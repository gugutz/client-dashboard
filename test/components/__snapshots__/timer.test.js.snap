// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> component should render correctly 1`] = `
<div>
  <h1
    className="title"
  >
    Timer
  </h1>
  <p>
    Seconds:
    12
  </p>
  <button
    onClick={[MockFunction]}
    type="button"
  >
    Reset
  </button>
</div>
`;
